v2-components/main-action-card/MainActionCard.tsx

```diff
@@ -0,0 +1,237 @@
+import { MaintainIcon } from '../icons/MaintainIcon/MaintainIcon';
+import { StakeIcon } from '../icons/StakeIcon/StakeIcon';
+
+const CardHeader = ({
+  step,
+  headingText,
+  bodyText,
+  icon,
+}: PropsWithChildren<{ step: number; headingText: string; bodyText: string; icon: ReactNode }>) => {
+  return (
+    <Box>
+      <Flex alignItems={'center'} justifyContent={'space-between'}>
+        <Text fontSize="4xl" fontFamily="mono">
+          {step}
+        </Text>
+        {icon}
+      </Flex>
+      <Heading fontSize={'sm'}>{headingText}</Heading>
+      <Text color={'whiteAlpha.700'} fontSize={'xs'}>
+        {bodyText}
+      </Text>
+    </Box>
+  );
+};
+
+const Container = ({ children }: PropsWithChildren<{}>) => {
+  return (
+    <Flex
+      flexDirection="column"
+      justifyContent="space-between"
+      width={['full', '70%', '72']}
+      height={['auto', 'auto', '56']}
+      alignItems="space-between"
+      border={'1px'}
+      borderColor="gray.800"
+      padding="2"
+      borderRadius="sm"
+    >
+      {children}
+    </Flex>
+  );
+};
+
+const StakeActionCard: React.FC<Props> = ({ isStaking }) => {
+  return (
+    <Container>
+      <CardHeader
+        step={1}
+        headingText="Stake & Borrow"
+        bodyText="Mint sUSD by staking your SNX."
+        icon={<StakeIcon />}
+      />
+      {isStaking ? (
+        <Button
+          onClick={() => {
+            console.log('navigate to stake and borrow more');
+          }}
+          mb="2"
+          variant="link"
+        >
+          {'Stake & Borrow More'}
+        </Button>
+      ) : (
+        <Button
+          onClick={() => {
+            console.log('navigate to stake and borrow more');
+          }}
+          variant={'solid'}
+        >
+          Start Staking
+        </Button>
+      )}
+    </Container>
+  );
+};
+const MaintainActionCard: React.FC<Props & { isFlagged: boolean }> = ({
+  liquidationCratioPercentage,
+  targetCratioPercentage,
+  currentCRatioPercentage,
+  isStaking,
+  isFlagged,
+}) => {
+  const variant = getVariant({
+    liquidationCratioPercentage,
+    targetCratioPercentage,
+    currentCRatioPercentage,
+  });
+
+  return (
+    <Container>
+      <CardHeader
+        step={2}
+        headingText="Maintain Collateralization Ratio"
+        bodyText="Maintain your Collateralization Health."
+        icon={<MaintainIcon />}
+      />
+      {isStaking && (
+        <Badge
+          color={variant}
+          bg="blackAlpha.600"
+          border={'1px'}
+          borderColor={variant}
+          display="flex"
+          alignItems={'center'}
+          width="fit-content"
+          margin={'0 auto'}
+          fontSize="x-small"
+          borderRadius={'base'}
+          fontWeight="700"
+        >
+          <InfoIcon color={variant} width="10px" height="10px" />
+          <Text ml="0.5">
+            {variant !== 'success'
+              ? 'Adjust to collect weekly rewards'
```

Comment:
Should this be in i18 json?

Reply:
You caught me. My other components are also missing i18.